lib/internal/repl/import.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const acorn = require('internal/deps/acorn/dist/acorn');
+const ESLoader = require('internal/loader/singleton');
+
+const kSymbolKey = 'internalReplLoaderPromise';
+const kInternalLoaderPromise = Symbol.for(kSymbolKey);
+
+/*
+Performs importing from static import statments
+
+"import { x, y } from 'some_modules'; x(); y"
+
+would become:
+
+(async () => {
+  const s = Symbol.for('internalReplLoaderPromise');
+  await this[s];
+  delete this[s];
+})().then(() => { x(); return (y); });
+
+The symbol is a promise from Promise.all on an array of promises
+from Loader#import. When a promise finishes it attaches to
+`repl.context` with a getter, so that the imported names behave
+like variables not values (live binding).
+*/
+
+
+function processTopLevelImport(src, repl) {
+  let root;
+  try {
+    root = acorn.parse(src, { ecmaVersion: 8, sourceType: 'module' });
+  } catch (err) {
+    return null;
+  }
+
+  const importLength =
+    root.body.filter((n) => n.type === 'ImportDeclaration').length;
+
+  if (importLength === 0)
+    return null;
+
+  const importPromises = [];
+  let newBody = '';
+
+  for (const index in root.body) {
```

Comment:
Using `for`-`in` to iterate over an array is risky since arrays can have other properties besides numeric indexes. In this case, it looks like you can just use a `forEach` loop:
```js
root.body.forEach((node, index) => {
  ...
});
```

Reply:
i trust acorn to not mess with things like this, but if you think its needed i can make the change

Reply:
use this, poison to Array.prototype with enumerable properties would also be picked up:

```
> Array.prototype.foo = 'bar';
'bar'
> for (var k in []) console.log(k)
foo
undefined
```